near-plugins-derive/src/access_control_role.rs

```diff
@@ -58,10 +58,10 @@ pub fn derive_access_control_role(input: TokenStream) -> TokenStream {
         ident, variants, ..
     } = input;
 
-    let variants = variants.into_iter().collect::<Vec<_>>();
+    let variant_idents = variants.into_iter().map(|v| v.ident).collect::<Vec<_>>();
```

Comment:
It was a bug to _not_ do `map(|v| v.ident)`. The test contract used only plain variants like `LevelA` without comments, data or custom discriminants, which prevented the bug from surfacing.